File: code/bit_manipulation/src/lonely_integer/lonely_integer.java
/*
* Part of Cosmos by OpenGenus Foundation
* The Lonely Integer Problem
* Given an array in which all the no. are present twice except one, find that lonely integer.
*/

public class LonelyInt{
  public static void main(String args[]){
    int[] testArr = {1,1,2,4,2};
    System.out.println(lonelyInt(testArr));
  }
  public static int lonelyInt(int[] arr){
    int lonely = 0;
    for(int i : arr){
        lonely ^= i;
    }
    return lonely;
  }
}

File: code/bit_manipulation/src/lonely_integer/lonely_integer.js
/*
 * Part of Cosmos by OpenGenus Foundation
 * The Lonely Integer Problem
 * Given an array in which all the no. are present twice except one, find that lonely integer.
 */
function lonelyInt(arr) {
  var lonely = 0;
  arr.forEach(number => {
    lonely ^= number;
  });
  return lonely;
}

function test() {
  var test = [1, 1, 2, 4, 2];
  console.log(lonelyInt(test));
}

test();
